tests/std/tests/VSO_0000000_c_math_functions/test.compile.pass.cpp

```diff
@@ -0,0 +1,12 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#if BOTH
+#include "both.cpp"

```

Comment:
Defining varying macros is a great technique. I think we should go further and simply fuse these `.cpp` and `.hpp` files into `test.compile.pass.cpp`, since we don't need separate TUs/headers now.